etna/whatson/models.py

```diff
@@ -0,0 +1,633 @@
+
+from etna.articles.models import ArticleTagMixin
+from etna.collections.models import TopicalPageMixin
+from etna.core.models import BasePageWithIntro
+
+
+class VenueType(models.TextChoices):
+    """
+    This model is used to add venue types to event pages.
+    """
+
+    ONLINE = "online", _("Online")
+    IN_PERSON = "in_person", _("In person")
+    HYBRID = "hybrid", _("Hybrid")
+
+
+@register_snippet
+class EventType(models.Model):
+    """
+    This snippet model is used so that editors can add event types,
+    which we use via the event_type ForeignKey to add event types
+    to event pages.
+    """
+
+    name = models.CharField(
+        max_length=255,
+        verbose_name=_("name"),
+        help_text=_("The name of the event type."),
+    )
+
+    class Meta:
+        verbose_name = _("Event type")
+        verbose_name_plural = _("Event types")
+
+    def __str__(self):
+        return self.name
+
+
+@register_snippet
+class AudienceType(models.Model):
+    """
+    This snippet model is used so that editors can add audience types,
+    which we use via the audience_type ForeignKey to add audience types
+    to event pages.
+    """
+
+    name = models.CharField(
+        max_length=255,
+        verbose_name=_("name"),
+        help_text=_("The name of the audience type."),
+    )
+
+    class Meta:
+        verbose_name = _("Audience type")
+        verbose_name_plural = _("Audience types")
+
+    def __str__(self):
+        return self.name
+
+
+class AudienceTypeOrderable(Orderable):
+    """
+    This model is used to add multiple audience types to event pages.
+    """
+
+    page = ParentalKey(
+        "wagtailcore.Page",
+        on_delete=models.CASCADE,
+        related_name="event_audience_types",
+    )
+
+    audience_type = models.ForeignKey(
+        "whatson.AudienceType",
+        on_delete=models.CASCADE,
+        related_name="audience_types",
+    )
+
+
+@register_snippet
+class AccessType(models.Model):
+    """
+    This snippet model is used so that editors can add access types,
+    which we use via the AccessTypeOrderable to add multiple access
+    types to event pages.
+    """
+
+    name = models.CharField(
+        max_length=255,
+        verbose_name=_("name"),
+        help_text=_("The type of access descriptor"),
+    )
+
+    class Meta:
+        verbose_name = _("Access type")
+        verbose_name_plural = _("Access types")
+
+    panels = [
+        FieldPanel("name"),
+    ]
+
+    def __str__(self):
+        return self.name
+
+
+class AccessTypeOrderable(Orderable):
+    """
+    This model is used to add multiple access types to event pages.
+    """
+
+    page = ParentalKey(
+        "wagtailcore.Page",
+        on_delete=models.CASCADE,
+        related_name="event_access_types",
+    )
+
+    access_type = models.ForeignKey(
+        "whatson.AccessType",
+        on_delete=models.CASCADE,
+        related_name="access_types",
+    )
+
+
+class Host(Orderable):
```

Comment:
Is there ever more than one host? Maybe these fields should be added directly to the page model instead (in which case, you would keep the `host_` prefix on the fields)

Reply:
I wondered this and apparently we can have more than one host, so I've put it in as an Orderable here. I'll remove the prefixes for most of the models, bad habit sorry! 😬